pyproject.toml

```diff
@@ -158,3 +148,45 @@ platforms = ["linux-64", "osx-64", "osx-arm64", "win-64", "osx-arm64", "linux-aa
 # This is necessary to pull in the lockfile/filelock dependency
 # since we don't handle the optional dependency.
 cachecontrol-with-filecache = ">=0.12.9"
+
+
+[tool.ruff]
+target-version = "py38"
+ignore = [
+    "E501",
+    "F401",
+    "F403",
+    # Disabled during migration to Ruff:
```

Comment:
Many of these can be enabled and autofixed, but I felt it would be better to tackle it in a separate PR to keep a reasonable scope for this one :)